packages/grid/_modules_/grid/hooks/features/filter/gridFilterSelector.ts

```diff
@@ -12,15 +12,16 @@ export const visibleGridRowsStateSelector = (state: GridState) => state.visibleR
 export const visibleSortedGridRowsSelector = createSelector<
   GridState,
   VisibleGridRowsState,
-  GridRowModel[],
-  GridRowModel[]
->(
-  visibleGridRowsStateSelector,
-  sortedGridRowsSelector,
-  (visibleRowsState, sortedRows: GridRowModel[]) => {
-    return [...sortedRows].filter((row) => visibleRowsState.visibleRowsLookup[row.id] !== false);
-  },
-);
+  Record<GridRowId, GridRowModel>,
+  Record<GridRowId, GridRowModel>
+>(visibleGridRowsStateSelector, sortedGridRowsSelector, (visibleRowsState, sortedRows) => {
+  return Object.keys(sortedRows).reduce((res, id) => {
+    if (visibleRowsState.visibleRowsLookup[id] !== false) {
+      res[id] = sortedRows[id];
+    }
+    return res;
+  }, {} as Record<GridRowId, GridRowModel>);
```

Comment:
Here we want to return the sorted rows array? Not sure why you refactored?

Reply:
Since we don't add an `id` to the row object, it's not reliable to use `row.id` anymore. I converted this selector to return an object because in this way we keep the reference to the original row as value and the keys are the correct ids.